test/vmtests/vmtests/imagecreator/test_imagecreator.py

```diff
@@ -185,36 +190,95 @@ def run_image_creator_test(
     logging.info(f"Attempting to connect to VM via SSH")
     logging.info(f"Username: {username}")
 
-    with vm.create_ssh_client(ssh_private_key_path, test_temp_dir, username) as ssh_client:
-        logging.info(f"SSH connection established successfully!")
-        # Run the test
-        logging.info(f"Running basic checks on the VM")
-        run_basic_checks(ssh_client, test_temp_dir)
-        logging.info(f"Basic checks completed successfully!")
+    try:
+        with vm.create_ssh_client(ssh_private_key_path, test_temp_dir, username) as ssh_client:
+            logging.info("SSH connection established successfully!")
+            # Test SSH connectivity and shell access
+            result = ssh_client.run("whoami")
+            if result.exit_code != 0:
+                logging.error("Failed to execute basic SSH command")
+                logging.error(f"Command output: {result.stdout}")
+                logging.error(f"Command error: {result.stderr}")
+                raise RuntimeError("SSH connection test failed")
+
+            # Run the test
+            logging.info("Running basic checks on the VM")
+            run_basic_checks(ssh_client, test_temp_dir)
+            logging.info("Basic checks completed successfully!")
+    except Exception as e:
+        logging.error(f"Failed to establish or verify SSH connection: {str(e)}")
+        logging.error("VM Console output:")
+        with open(vm_console_log_file_path, "r") as f:
+            logging.error(f.read())
+        raise
 
 
 def run_basic_checks(
     ssh_client: SshClient,
     test_temp_dir: Path,
 ) -> None:
+    """Run basic checks on the VM to verify the OS installation.
 
+    The checks are distro-specific based on the os-release file content.
+    """
     ssh_client.run("cat /proc/cmdline").check_exit_code()
 
     os_release_path = test_temp_dir.joinpath("os-release")
     ssh_client.get_file(Path("/etc/os-release"), os_release_path)
 
     with open(os_release_path, "r") as os_release_fd:
         os_release_text = os_release_fd.read()
-
-        # Since imagecreator creates new images, we expect Azure Linux 3.0
-        assert "ID=azurelinux" in os_release_text
-        assert 'VERSION_ID="3.0"' in os_release_text
-
-    # Check that essential packages are installed
-    ssh_client.run("rpm -q kernel").check_exit_code()
-    ssh_client.run("rpm -q systemd").check_exit_code()
-    ssh_client.run("rpm -q grub2").check_exit_code()
-    ssh_client.run("rpm -q bash").check_exit_code()
+        logging.info("OS Release content:")
+        logging.info(os_release_text)
+
+        # Determine which distro we're testing
```

Comment:
Don't do dynamic checking in tests. It can sometimes hide errors when things go wrong. Instead, the test should know in advance which image it is testing.